charts/newrelic-k8s-metrics-adapter/tests/deployment_test.yaml

```diff
@@ -0,0 +1,106 @@
+release:
+  name: my-release
+  namespace: my-namespace
+tests:
+  - it: has the correct image
+    set:
+      personalAPIKey: 21321
+      image:
+        repository: newrelic/newrelic-k8s-metrics-adapter
+        tag: "latest"
+        pullSecrets:
+          - name: regsecret
+      config:
+        accountID: 111
+        region: A-REGION
+    asserts:
+      - matchRegex:
+          path: spec.template.spec.containers[0].image
+          pattern: ^.*newrelic/newrelic-k8s-metrics-adapter:latest
+        template: templates/deployment.yaml
+      - equal:
+          path: spec.template.spec.imagePullSecrets
+          value:
+            - name: regsecret
+        template: templates/deployment.yaml
+  - it: correctly uses the cluster helper
+    set:
+      personalAPIKey: 21321
+      config:
+        accountID: 111
+        region: A-REGION
+      cluster: a-cluster
+    asserts:
+      - equal:
+          path: spec.template.spec.containers[0].env[0].value
+          value: a-cluster
+        template: templates/deployment.yaml
+  - it: correctly uses common.securityContext.podDefaults
+    set:
+      personalAPIKey: 21321
+      config:
+        accountID: 111
+        region: A-REGION
+      cluster: a-cluster
+    asserts:
+      - equal:
+          path: spec.template.spec.securityContext
+          value:
+            fsGroup: 1001
+            runAsGroup: 1001
+            runAsUser: 1001
+        template: templates/deployment.yaml
+  - it: renders when extraEnv, extraEnvFrom, extraVolumes and extraVolumeMounts are set
+    set:
+      personalAPIKey: 21321
+      config:
+        accountID: 111
+        region: A-REGION
+      extraEnv:
+        - name: ENV_VAR1
+          value: "var1"
+        - name: ENV_VAR2
+          value: "var2"
+      extraEnvFrom:
+        - configMapRef:
+            name: special-config
+      extraVolumes:
+        - name: tmpfs-data
+          emptyDir: {}
+      extraVolumeMounts:
+        - mountPath: /var/db/newrelic-infra/data
+          name: tmpfs-data
+    asserts:
+      - hasDocuments:
+          count: 1
+        template: templates/deployment.yaml
+  - it: renders template with values for all helpers
+    set:
+      personalAPIKey: 21321
+      config:
+        accountID: 111
+        region: A-REGION
+      priorityClassName: high-priority
+      hostNetwork: true
+      dnsConfig:
+        keyOnValues: bar
+      tolerations:
+        - key: "key1"
+          operator: "Exists"
+          effect: "NoSchedule"
+      nodeSelector:
+        disktype: ssd
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+              - matchExpressions:
+                  - key: kubernetes.io/e2e-az-name
+                    operator: In
+                    values:
+                      - new
+    asserts:
+      - hasDocuments:
+          count: 1
+        template: templates/deployment.yaml
```

Comment:
NIT: I would remove this and add it to a file in `charts/newrelic-k8s-metrics-adapter/ci/whatever.yaml` so it would not only test that it would template but creates a valid manifest that is installable.

Sorry for do not think about this possibility earlier.